### DISCUSSION TITLE
Show branch when using workflow_dispatch

### DISCUSSION THREAD

**Discussion by nichmj01** [community/community#15690] [Actions] [answered]
Currently, when you trigger a workflow manually using `on workflow_dispatch`, the workflow doesn't tell you which branch it was run on, while `on push` and `on pull_request` runs do. It would be nice if `on workflow_dispatch` showed the branch just like those other event triggers. What's interesting is that the branch is known since I can filter by branch, it's just not shown on the list. 

Requests for this functionality are also mentioned in #13689 and #11396. I figured it would be useful to call it out specifically since it seems like this would be a bit easier to implement vs the larger request in those other discussions.

**Comment by gabrielrra** [comment]
I really think that would be great

**Reply by sualeh** [reply]
Yes, this is an important enhancement to be made.

**Reply by jbm00n** [reply]
The CLI does it, so would be great to have that info from the web UI as well

**Reply by muellerk22** [reply]
I seeing the same and it drive me crazy!

**Reply by CompRhys** [reply]
bump

**Reply by lmjhs** [reply]
Yeah this is sorely needed, please fix.

**Reply by armen-mobeus** [reply]
Azure DevOps already does this, why GHA doesn't?

**Reply by Lukerayn3r** [reply]
Has this been added yet?

**Reply by mBlomsterberg** [reply]
It would be nice to see this introduced soon. Difficult to distinguish what workflow ran on which branch.

**Reply by nnellans** [reply]
This seems like pretty basic functionality that should be implemented.  Why this is taking so long I have no idea.

**Reply by ajsharma12541** [reply]
+1

**Reply by dmalyshevmarinexchange** [reply]
Looking forward to this feature!

**Reply by ghost** [reply]
+1

**Reply by babismylonas** [reply]
This is still not implemented? Indeed it's pretty basic functionality and also very important, since you can't distinguish on which branch the same workflow ran in the past!

**Comment by shobhit-tiwari-maersk** [comment]
+1 really hard to figure out which branch was used

**Comment by ghost** [comment]
Even more, some actions from market use the target branch name for it's work but, as manual dispatches, even though they ask u which branch to target, does not provide that info on the event (or the way other triggers do) makes some actions fail or be less worthy,.

Specific case is a code analysis tool wich sends reports to third party site (their own), missing the branch name makes the report useless there.

**Comment by dweller23** [comment]
A partial solution would be leverage `run-name` introduced almost a year ago: https://github.blog/changelog/2022-09-26-github-actions-dynamic-names-for-workflow-runs/

According to the docs over here:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name

you can pass `github` context variable, and you can use `ref` value there. Not the prettiest formatting, but does the job. 

So an example would be having something like that in the top level of the workflow:
```
run-name: Manually triggered workflow from ${{ github.ref }}
```

**Reply by jbm00n** [reply]
I'd prefer a proper integration from GH UI but this is a nice workaround, thanks!

I use ${{ github.ref_name }} for a shorter branch name (without the refs/heads part)

**Reply by nichmj01** [reply]
Thanks for the workaround suggestion! Not an ideal solution but reasonable.

**Reply by ghost** [reply]
This works, however it means adding more information to a run name that may already be heavily populated.
eg we have a reusable workflow where the run-name already contains environment to deploy to, deployment type, and component type. Making the run name long enough as it is!
Seems strange that there is a column for the branch, and then not populate that column

**Reply by lmjhs** [reply]
I agree @jbruce-nex, it's worth noting that the branch filter works correctly so the underlying data is there. Feels like it is just a visual bug where the branch is just not shown for some reason, hoping there may be a very easy fix coming in future.

**Comment by mehulrajdev** [comment]
Hello! 👋
Thanks for the details. Does the billing apply yearly or monthly?

**Comment by anuragsati** [comment]
This is still not implemented, any updates on this?

**Comment by sirtyman** [comment]
For our case the lack of the branch associated with manually triggered workflow causes another issue. When a PR is opened by dependabot, that in turn doesn't have access to secrets, auto-triggered workflows fail. We thought manual trigger is solution for re-running workflows against branch created by dependabot. This is possible. However, these runs, as they have no association with the specific branch, are not being attached to the specific PRs. The only solution in such case is the make fake commit on dependabot branch to re-trigger automatic workflows. Really annoying.

**Comment by andrewakim** [comment, marked as answer]
:wave: Hi everyone. Branch names should now show up on workflow runs triggered via `workflow_dispatch`!

**Reply by anuragsati** [reply]
Yes they show up now! Thanks 😄

**Reply by Niyaskva** [reply]
where is the answer?

**Reply by jpcmonster** [reply]
What about GHE? :)  I don't see it in 3.10.3 :/

**Comment by nandahibatullah** [comment]
Hey there, how do I get it to appear for workflow runs triggered via the release event?
